sdk/servicebus/azure-servicebus/tests/perf_tests/_test_base.py

```diff
@@ -5,124 +5,307 @@
+        )
+        self.topic_name=self.get_from_env('AZURE_SERVICEBUS_TOPIC_NAME')
+        self.subscription_name=self.get_from_env('AZURE_SERVICE_BUS_SUBSCRIPTION_NAME')
+
+        self.receiver=self.servicebus_client.get_subscription_receiver(topic_name=self.topic_name, subscription_name=self.subscription_name)
+        self.async_receiver=self.async_servicebus_client.get_subscription_receiver(topic_name=self.topic_name, subscription_name=self.subscription_name)
+    
+    async def global_setup(self) -> None:
+        await super().global_setup()
+        await self._preload_topic()
+    
+    async def close(self) -> None:
+        self.receiver.close()
+        await self.async_receiver.close()
+        await self.close_clients()
+        await super().close()
+    
+    @staticmethod
+    def add_arguments(parser) -> None:
+        super(_SubscriptionReceiveTest, _SubscriptionReceiveTest).add_arguments(parser)
+        parser.add_argument('--message-size', nargs='?', type=int, help='Size of a single message. Defaults to 100 bytes', default=100)
+        parser.add_argument('--num-messages', nargs='?', type=int, help='Maximum number of messages to receive. Defaults to 100', default=100)
+        parser.add_argument('--peeklock', action='store_true', help='Receive using PeekLock mode and message settlement.', default=False)
+        parser.add_argument('--uamqp-transport', action="store_true", help="Switch to use uamqp transport. Default is False (pyamqp).", default=False)
+        parser.add_argument('--transport-type', nargs='?', type=int, help="Use Amqp (0) or Websocket (1) transport type. Default is Amqp.", default=0)
+        parser.add_argument('--max-wait-time', nargs='?', type=int, help='Max time to wait for messages before closing. Defaults to 0.', default=0)
+        parser.add_argument('--preload', nargs='?', type=int, help='Number of messages to preload. Default is 10000.', default=10000)
+
+
+class _QueueReceiveBatchTest(_ReceiveTest, BatchPerfTest):
+    def __init__(self, arguments) -> None:
+        super().__init__(arguments)
+        self.setup_servicebus_clients(
+            arguments.transport_type, 
+            arguments.peeklock, 
+            arguments.num_messages, 
+            arguments.max_wait_time, 
+            arguments.uamqp_transport
+        )
+        self.queue_name=self.get_from_env('AZURE_SERVICEBUS_QUEUE_NAME')
+        self.receiver=self.servicebus_client.get_queue_receiver(self.queue_name)
+        self.async_receiver=self.async_servicebus_client.get_queue_receiver(self.queue_name)
+    
+    async def global_setup(self) -> None:
         await super().global_setup()
         await self._preload_queue()
     
-    async def close(self):
+    async def close(self) -> None:
         self.receiver.close()
         await self.async_receiver.close()
+        await self.close_clients()
         await super().close()
+    
+    @staticmethod
+    def add_arguments(parser) -> None:
```

Comment:
It looks like this `add_arguments` method could be moved up onto the mixin, as it seems to be generally the same for all the receive tests?

Reply:
I had it set up like that initially but the call to super was erroring out as its static. Given that none of the other classes have add arguments, I dont need to call it :)